SW/Boiling Detection/boil_detect.py

```diff
@@ -0,0 +1,240 @@
+        if self.first_image is None:
+            self.first_image = frame
+            self.first_quads = self.__split_frame(self.first_image)
+
+        # if the number of frames that have been processed are desired amount, proceed
+        elif self.frame_count == self.count:
+            # if there is no second image recorded yet
+            if self.second_image is None:
+                self.second_image = frame
+                self.second_quads = self.__split_frame(self.second_image)
+            # second image does exist, must shift images
+            else:
+                self.first_image = self.second_image
+                self.second_image = frame
+
+            # splits the frames
+            self.first_quads = self.__split_frame(self.first_image)
+            self.second_quads = self.__split_frame(self.second_image)
+
+            # Detects boiling on active burners
+            i = 0
+            for burner in burners:
+                # if there is a pot on burner
+                if burner.pot_detected:
+                    mask = self.masks[i]
+                    score = self.___get_score(mask, self.first_quads[i], self.second_quads[i])
+                    self.median_SSIM[i].append(score)
+
+                    # median is calculated when enough SSIMs have been recorded
+                    if len(self.median_SSIM[i]) == self.SSIM_count:
+                        med = np.median(self.median_SSIM[i])
+                        self.boiling[i].append(med)
+                        self.median_SSIM[i].clear()
+
+                    # ensures that the boiling list is not empty before trying to calculate status
+                    if len(self.boiling[i]) > 0:
+                        # waits to set boiling flag until 30 seconds has passed and the began_boiling flag remains true
+                        if self.began_boiling[i] and time.time() - self.start_boiling[i] >= 30:
+                            burner.boiling = True
+                        elif self.began_boiling[i]:
+                            self.began_boiling[i] = self.__check_boiling(self.boiling[i], self.thresh)
+                            # if beganBoiling becomes false, resets the flag
+                            if not self.began_boiling[i]:
+                                self.start_boiling[i] = time.time()  # sets time to when it began boiling
+                        else:
+                            self.began_boiling[i] = self.__check_boiling(self.boiling[i], self.thresh)
+                            self.start_boiling[i] = time.time() # keeps track of time when started boiling
+                            if self.began_boiling[i] or time.time() >= 30:
+                                self.boiling[i].clear()  # clear the boiling list for new values to be collected
+                i = i + 1
+
+        # sets appropriate stove burners to boiling
+        if burners[0].boiling and stove.upper_left.temp >= self.temp:
```

Comment:
I think this can be refactored to foreach(burner in stove.get_burners()) (once you implement a get_burners function, like mentioned above)

Reply:
Question: if I have the following code,

        for burner, burner_boil in zip(burners,self.boiling):
            if burner_boil and burner.temp >= self.MIN_BOILING_TEMP:
                burner.boiling = True 
            else:
                burner.boiling = False 

will the "burner.boiling" change the actual value of the stove burner? not just the "burner" object being used to iterate?